src/services/shims.ts

```diff
@@ -322,8 +323,19 @@ module ts {
         }
     }
 
+    function serializeResultAsJson(target: Shim, key: string | symbol, descriptor: PropertyDescriptor) {
+        let action = descriptor.value;
+        descriptor.value = function () {
+            let _arguments = arguments;
+            return forwardJSONCall((<Shim>this).logger,
+                <string>key + "(" + JSON.stringify(Array.prototype.join.call(_arguments, ", ")) + ")",
+                () => action.apply(this, _arguments));

```

Comment:
I noticed that you use `this` in two places here. Is it the same as `target`? If so, I think I'd rather use `target`.